soc/xtensa/intel_adsp/tools/remote-fw-service.py

```diff
@@ -143,7 +143,8 @@ def handle(self):
 
         log.info("FW is ready...")
 
-        with subprocess.Popen(runner.get_script(), stdout=subprocess.PIPE) as proc:
+        with subprocess.Popen(runner.get_script(), stdout=subprocess.PIPE,
+                              start_new_session=True) as proc:
```

Comment:
aren't you creating a new session leader for the new process ? I thought the current process should be the session leader for the child process, so if it closes the child process would receive `SIGHUP`

Reply:
This was also a little confusing to me as well, though the explanation here is because the child is being run with sudo when remote-fw-service itself is run as a normal user (makes sense, its on a socket, etc) that this is needed for sudo to accept the signals.

https://man7.org/linux/man-pages/man8/sudo.8.html Process Model section and Signal Handling section give some details on it.

I'm not super familiar with this but everything in the PR description lines up it seems.